src-java/ping-topology/ping-storm-topology/src/main/java/org/openkilda/wfm/topology/ping/bolt/OnDemandResultManager.java

```diff
@@ -180,7 +180,9 @@ private YFlowPingResponse buildYFlowPingResponse(Group group) {
                     "Expect exact one Y Flow id in pings group response, got - %s", yFlowIdSet));
         }
 
-        return new YFlowPingResponse(yFlowIdSet.iterator().next(), null, new ArrayList<>(subFlowMap.values()));
+        String error = (subFlowMap.size() & 1) == 1 ? "One sub_flow is one-switch flow" : null;
```

Comment:
> (subFlowMap.size() & 1) == 1
What is the purpose of this hack? I'd recommend either to move it to private method with meaningful name like 
``` boolean isAnySubflowIsOneSwitch(Map<SubFlow> subFlowMap)``` or to rewrite it without bitshifting. I doubt that the person who will read your code is able to do bitshifting in mind to understand what could be possible cases.

Reply:
This is just an optimised way to check if the size is odd. I was initially using an if instead of a ternary operator and I was including a comment. And, yes, I agree that it is not so readable. I will change it. Thanks